tools/nebula-flink/src/main/java/org.apache.flink/connector/nebula/sink/NebulaSinkFunction.java

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package org.apache.flink.connector.nebula.sink;
+
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.runtime.state.CheckpointListener;
+import org.apache.flink.runtime.state.FunctionInitializationContext;
+import org.apache.flink.runtime.state.FunctionSnapshotContext;
+import org.apache.flink.streaming.api.checkpoint.CheckpointedFunction;
+import org.apache.flink.streaming.api.functions.sink.RichSinkFunction;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class NebulaSinkFunction<T> extends RichSinkFunction<T> implements CheckpointedFunction {
+
+    private static final long serialVersionUID = 8100784397926666769L;
+
+    private final AbstractNebulaOutPutFormat<T> outPutFormat;
+
+    private final AtomicReference<Throwable> failureThrowable = new AtomicReference<>();
```

Comment:
why defined as atomic variable ？ this one will happen conflict?

Reply:
It was originally intended to be used in NebulaBatchExecutor.executeBatch's callback function, currently I only print error log when throwable occurs.